### DISCUSSION TITLE
Possible to redirect?

### DISCUSSION THREAD

**Discussion by ndarilek** [benallfree/pocketpages#11] [Q&A] [answered]
I'm prototyping an app that manages templates. If someone hits / with no templates, I'd like to redirect to /templates/new. Is that possible?

My assumption was that I could do this in middleware, but while it seems I can throw an error, I can't redirect the request. Am I missing something?

Also, as an aside, are there any sample apps I can look at? I'm not entirely clear what best practices are for, say, loading a record from an ID and returning 404 if it isn't found, if it's possible to change the default 404 page, etc.

Thanks again.

**Comment by benallfree** [comment, marked as answer]
`ctx.redirect(status, path)` should work.

I'm about to release v10 which has a `redirect()` helper and some code samples too. But in the mean time, `ctx.redirect` should work.

There will be lots more samples in v10. 

Good question about a custom 404. Do you know if PocketBase allows 404 customization? I assume it does, but I've never looked.

**Reply by ndarilek** [reply]
Oh nice, just pulled the repo and am looking at the new docs. One thing I also wondered about once I asked this: POST handling. Not sure if I'll stick with pocketpages forever or eventually switch to a more standard SPA framework, but my thought for now is to do all core logic in Go and use pocketpages for the UI. I'm guessing I can handle form posts in middleware by checking the method and adding my form validation/record creation logic there? Just wanted to make sure that creating directories for pages/middleware also intercepted events other than GET at those paths.

Thanks again, really digging this project.

**Reply by benallfree** [reply]
I would suggest implementing the core logic in JS as well.

Can you please give me a more concrete example of how you plan to expose Go customizations to PocketPages?

**Comment by ndarilek** [comment]
> Can you please give me a more concrete example of how you plan to expose Go customizations to PocketPages?

By using record/model hooks. So if someone creates a template in my templates collection, or updates it, create/update hooks make a POST request to Docuseal, sending it a link to the file. Docuseal then ingests the template, processes it into a template I can then use to request signatures, responds with its internal template ID, and the record in PB gets updated. All PocketPages has to do is create the appropriate records and manage the UI. [Here's a blog post I wrote for a previous project](https://fly.io/blog/llm-image-description/) showing off the approach more directly, albeit using a Python module rather than a web frontend.

**Reply by benallfree** [reply]
Right, I'd still do that in JS rather than Go but that's just because I don't want to constantly recompile and maintain a custom PocketBase build.
